### PR TITLE
[UI] Show step pod yaml and events in RunDetails page

### PR DESCRIPTION
**Pull request by Bobgy**
/area frontend
/kind feature

**Demo for pod yaml**: https://youtu.be/HvsPvUEW4Kk
**Demo for pod events**: https://youtu.be/R-zdTEwoNmw

Resolves https://github.com/kubeflow/pipelines/issues/2094
Resolves https://github.com/kubeflow/pipelines/issues/1711
Resolves https://github.com/kubeflow/pipelines/issues/3112

TODO:
* [x] add related unit tests

Extra changes in this PR:
* I wrote new react component tests using `@testing-library/react`: https://github.com/kubeflow/pipelines/pull/3304/files#diff-0fc4f908858a81da473a9fcae07bcd30.
This the first PR with a very comprehensive test using `@testing-library/react`: https://github.com/kubeflow/pipelines/issues/2732.
* Refactored k8s-helper to no longer export `inCluster` property, so that I could unit test newly added handlers that use k8s-helper: https://github.com/kubeflow/pipelines/pull/3304/files#diff-98ad2c53495f413ec48e3835fc45b86eL55
Resolves https://github.com/kubeflow/pipelines/issues/2215

I personally think `@testing-library/react` is much easier to use comparing to enzyme, because of
* helper methods to get element by text
* helper methods to fire events
* it doesn't test any implementation detail by default (no shallow testing), always asserts the dom
* rerender helper to test a component when its props change

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/3304)

### CODE REVIEW

File: frontend/server/k8s-helper.ts [hunk 1]
```diff
@@ -168,6 +162,9 @@ export async function getTensorboardInstance(
  */
 
 export async function deleteTensorboardInstance(logdir: string): Promise<void> {
+  if (!namespace) {
+    throw new Error(`Cannot get namespace from ${namespaceFilePath}.`);
+  }
   if (!k8sV1CustomObjectClient) {
```
Comment (jingzhang36) [inline frontend/server/k8s-helper.ts]: In other places, the checks on custom object client are removed, can we also remove the check here?

Comment (Bobgy) [reply frontend/server/k8s-helper.ts]: Thanks! Yes, we can also remove this

Comment (Bobgy) [reply frontend/server/k8s-helper.ts]: Fixed

**Pull request merged**